src/executor/executor.c

```diff
@@ -0,0 +1,252 @@
+
+                        (void) fd_cloexec(fd, true);
+
+                        f = fdopen(fd, "r");
+                        if (!f)
+                                return log_error_errno(errno, "Failed to open serialization fd %d: %m", fd);
+
+                        safe_fclose(arg_serialization);
+                        arg_serialization = f;
+
+                        break;
+                }
+
+                case '?':
+                        return -EINVAL;
+
+                default:
+                        assert_not_reached();
+                }
+
+        if (!arg_serialization)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "No serialization fd specified.");
+
+        return 1 /* work to do */;
+}
+
+int main(int argc, char *argv[]) {
+        _cleanup_fdset_free_ FDSet *fdset = NULL;
+        _cleanup_(unit_freep) Unit *unit = NULL;
+        int exit_status = EXIT_SUCCESS, r;
+        _cleanup_(exec_params_serialized_clear) ExecParameters params = {
+                .stdin_fd         = -EBADF,
+                .stdout_fd        = -EBADF,
+                .stderr_fd        = -EBADF,
+                .exec_fd          = -EBADF,
+                .user_lookup_fd   = -EBADF,
+                .bpf_outer_map_fd = -EBADF,
+        };
+        ExecCommand command = {};
+        DynamicCreds dynamic_creds = {
+        };
+        ExecSharedRuntime shared = {
+                .netns_storage_socket = PIPE_EBADF,
+                .ipcns_storage_socket = PIPE_EBADF,
+        };
+        ExecRuntime runtime = {
+                .ephemeral_storage_socket = PIPE_EBADF,
+                .shared = &shared,
+                .dynamic_creds = &dynamic_creds,
+        };
+
+        /* We might be starting the journal itself, we'll be told by the caller what to do */
+        log_set_always_reopen_console(true);
+        log_set_prohibit_ipc(true);
+        log_setup();
+
+        r = fdset_new_fill(/* filter_cloexec= */ 0, &fdset);
+        if (r < 0)
+                return log_error_errno(r, "Failed to create fd set: %m");
+
+        r = parse_argv(argc, argv);
+        if (r <= 0)
+                return r;
+
+        /* Now try again if we were told it's fine to use a different target */
+        if (log_get_target() != LOG_TARGET_KMSG) {
+                log_set_prohibit_ipc(false);
+                log_open();
+        }
+
+        r = fdset_remove(fdset, fileno(arg_serialization));
+        if (r < 0)
+                return log_error_errno(r, "Failed to remove serialization fd from fd set: %m");
+
+        r = exec_deserialize(arg_serialization, fdset, &unit, &command, &params, &runtime);
+        if (r < 0) {
+                exit_status = log_error_errno(r, "Failed to deserialize: %m");
+                goto out;
+        }
+
+        arg_serialization = safe_fclose(arg_serialization);
+        fdset = fdset_free(fdset);
+
+        r = exec_child(unit,
```

Comment:
hmm, we should find a different name for `exec_child()` now. Previously, it was called that way, because it is what we execute in the child of the code the function as embedded in. But here we already are the child, hence this is confusing, because to the casual onlooker it looks as if this would actually fork a child here. Except it really doesn't, there's no forking involved here anymore.

Hence maybe: exec_invoke()? exec_go()? exec_now()?